docs/pages/image-extractor.rst

```diff
@@ -0,0 +1,158 @@
+    support extracting images of 'unnavigable points' like walls. In the future we hope to extend this functionality to allow the user to specify more unique class labels, but for now this argument is not that useful.
+* img_size: The size of images to be output in the format (height, width)
+* output: A list of the different output image types the user can obtain. Default is rgba.
+
+Habitat Sim does not currently support multiple instances of extractors, so if you're done using
+an extractor you need to call the close method before instantiating a new one. Below we will go
+over some optional arguments for the extractor class.
+
+
+`Using the Extractor`_
+======================
+
+The extractor can be indexed and sliced like a normal python list. Internaly, indexing into
+the extractor sets an agent position and rotation within the simulator and returns the corresponding
+agent observation. Indexing returns a dictionary that contains an image of each type specified in
+the "output" argument given by the user in the constructor. The dictionary also contains a key
+"label" which is the class label (specified by the user in the constructor) of the image. Note:
+for the scene in this example there is no semantic data which is why the semantic output
+below is not represented.
+
+.. code:: py
+
+    from habitat_sim.utils.data.data_extractor import ImageExtractor
+
+    # Give the extractor a path to the scene
+    scene_filepath = "../data/scene_datasets/habitat-test-scenes/skokloster-castle.glb"
+
+    # Instantiate an extractor. The only required argument is the scene filepath
+    extractor = ImageExtractor(scene_filepath, labels=[0.0], img_size=(512, 512),
+                            output=['rgba', 'depth', 'semantic'])
+
+    # Index in to the extractor like a normal python list
+    sample = extractor[0]
+
+    # Or use slicing
+    samples = extractor[1:4]
+    for sample in samples:
+        display_sample(sample)
+
+
+.. image:: /static/extractor-example-output.png
+
+
+`Integrating with Pytorch Datasets`_
+====================================
+
+It is very easy to plug an ImageExtractor into a PyTorch Datasets and DataLoaders for end to end
+training in PyTorch models without writing to disk. For a great tutorial on how to use PyTorch Dataset
+and DataLoader, refer to this guide: https://pytorch.org/tutorials/beginner/data_loading_tutorial.html
+
+.. code:: py
+
+    import torch
+    import torch.nn as nn
+    import torch.nn.functional as F
+    from torch.utils.data import Dataset, DataLoader
+
+    class HabitatDataset(Dataset):
+        def __init__(self, extractor):
+            self.extractor = extractor
+
+        def __len__(self):
+            return len(self.extractor)
+
+        def __getitem__(self, idx):
+            sample = self.extractor[idx]
+            output = {
+                'rgba': sample['rgba'].astype(np.float32), # dataloader requires certain types
```

Comment:
```suggestion
                'rgb': sample['rgba'].astype(np.float32)[..., :-1], # dataloader requires certain types
```
Most people will probably want rgb, not rgba

Reply:
right but the color sensor outputs rgba so wouldn't it been more accurate to keep it as rgba? That was the same argument for changing the output name from "rgb" to "rgba" in data_extractor.py a while back.

Reply:
Naming it `rgba` was needed as this is rgba (it would have also been fine to convert rgba to rgb and then name it rgb).  Most people will want to train models to operate on rgb not rgba (that is the standard thing output by cameras) so it is useful to who them how to convert from rgb to rgba (it isn't strictly true that you can just chop off alpha to go from rgba to rgb, but it works in this case).